test/torchaudio_unittest/prototype/functional/functional_test_impl.py

```diff
@@ -149,6 +149,149 @@ def test_oscillator_invalid(self):
+                num_frames=11,
+                expected=[1.0, 0.8, 0.6, 0.4, 0.2, 0, 0, 0, 0, 0, 0],
+                decay=0.5,
+                n_decay=1,
+            ),
+            # Decay (partial - polynomial)
+            param(
+                num_frames=11,
+                expected=[1.0, 0.64, 0.36, 0.16, 0.04, 0, 0, 0, 0, 0, 0],
+                decay=0.5,
+                n_decay=2,
+            ),
+            # Decay (full - linear)
+            param(
+                num_frames=11,
+                expected=[1.0 - i / 10 for i in range(11)],
+                decay=1.0,
+                n_decay=1,
+            ),
+            # Decay (full - polynomial)
+            param(
+                num_frames=11,
+                expected=[(1.0 - i / 10) ** 2 for i in range(11)],
+                decay=1.0,
+                n_decay=2,
+            ),
+            # Sustain (partial - preceeded by decay)
+            param(
+                num_frames=11,
+                expected=[1.0, 0.9, 0.8, 0.7, 0.6, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5],
+                decay=0.5,
+                sustain=0.5,
+                n_decay=1,
+            ),
+            # Sustain (partial - preceeded by decay)
+            param(
+                num_frames=11,
+                expected=[1.0, 0.8, 0.6, 0.4, 0.4, 0.4, 0.4, 0.4, 0.4, 0.4, 0.4],
+                decay=0.3,
+                sustain=0.4,
+                n_decay=1,
+            ),
+            # Sustain (full)
+            param(
+                num_frames=11,
+                expected=[0.3] * 11,
+                sustain=0.3,
+            ),
+            # Release (partial - preceeded by decay)
+            param(
+                num_frames=11,
+                expected=[1.0, 0.84, 0.68, 0.52, 0.36, 0.2, 0.16, 0.12, 0.08, 0.04, 0.0],
+                decay=0.5,
+                sustain=0.2,
+                release=0.5,
+                n_decay=1,
+            ),
+            # Release (partial - preceeded by sustain)
+            param(
+                num_frames=11,
+                expected=[0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.4, 0.3, 0.2, 0.1, 0.0],
+                sustain=0.5,
+                release=0.5,
+            ),
+            # Release (full)
+            param(
+                num_frames=11,
+                expected=[1 - i / 10 for i in range(11)],
+                sustain=1.0,
+                release=1.0,
+            ),
+        ]
+    )
+    def test_adsr_envelope(
+        self, num_frames, expected, attack=0.0, hold=0.0, decay=0.0, sustain=0.0, release=0.0, n_decay=2.0
+    ):
+        """the distribution of time are correct"""
+        out = F.adsr_envelope(
+            num_frames,
+            attack=attack,
+            hold=hold,
+            decay=decay,
+            sustain=sustain,
+            release=release,
+            n_decay=n_decay,
+            device=self.device,
+            dtype=self.dtype,
+        )
+        print(out)
+        print(expected)
```

Comment:
remove the print statements

Reply:
It's useful to print relevant information when it fails, but sure